plugins/woocommerce-blocks/assets/js/base/components/button/index.tsx

```diff
@@ -1,70 +1,127 @@
-import type { Button as WPButtonType } from '@wordpress/components';
+import { Button as AriakitButton } from '@ariakit/react';
+import { forwardRef } from '@wordpress/element';
 import classNames from 'classnames';
+import type { ForwardedRef } from 'react';
+import type { ButtonProps as AriakitButtonProps } from '@ariakit/react';
+import deprecated from '@wordpress/deprecated';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
 import Spinner from '../../../../../packages/components/spinner';
 
-export interface ButtonProps
-	extends Omit< WPButtonType.ButtonProps, 'variant' | 'href' > {
+type WCButtonProps = AriakitButtonProps & { children?: React.ReactNode };
+
+export interface ButtonProps extends WCButtonProps {
 	/**
-	 * Show spinner
+	 * Deprecated: Show a spinner. Preferably,
+	 * render a spinner in the button children
+	 * instead.
 	 *
 	 * @default false
 	 */
 	showSpinner?: boolean | undefined;
 	/**
 	 * Button variant
+	 *
+	 * @default 'contained'
 	 */
 	variant?: 'text' | 'contained' | 'outlined';
 	/**
-	 * The URL the button should link to.
+	 * By default we render a wrapper around  the button children,
+	 * but you can opt in to removing it by setting removeTextWrap
+	 * to true.
+	 *
+	 * @default false
 	 */
-	href?: string | undefined;
+	removeTextWrap?: boolean;
 }
 
-export interface AnchorProps extends Omit< ButtonProps, 'href' > {
+interface LinkProps extends ButtonProps {
 	/**
 	 * Button href
 	 */
-	href?: string | undefined;
+	href: string;
 }
 
 /**
  * Component that visually renders a button but semantically might be `<button>` or `<a>` depending
  * on the props.
  */
-const Button = ( {
-	className,
-	showSpinner = false,
-	children,
-	variant = 'contained',
-	...props
-}: ButtonProps ): JSX.Element => {
-	const buttonClassName = classNames(
-		'wc-block-components-button',
-		'wp-element-button',
-		className,
-		variant,
-		{
-			'wc-block-components-button--loading': showSpinner,
+const Button = forwardRef< HTMLButtonElement, ButtonProps | LinkProps >(
+	( props, ref ) => {
+		if ( 'showSpinner' in props ) {
+			deprecated( 'showSpinner prop', {
+				version: '8.9.0',
+				alternative: 'Render a spinner in the button children instead.',
+				plugin: 'WooCommerce',
+			} );
 		}
-	);
 
-	return (
-		<WPButton className={ buttonClassName } { ...props }>
-			{ showSpinner && <Spinner /> }
+		const {
+			className,
+			showSpinner = false,
+			children,
+			variant = 'contained',
+			// To maintain backward compat we render a wrapper for button text by default,
+			// but you can opt in to removing it by setting removeTextWrap to true.
+			removeTextWrap = false,
+			...rest
+		} = props;
+
+		const buttonClassName = classNames(
+			'wc-block-components-button',
+			'wp-element-button',
+			className,
+			variant,
+			{
+				'wc-block-components-button--loading': showSpinner,
+			}
+		);
+
+		if ( 'href' in props ) {
+			return (
+				<AriakitButton
+					render={
+						<a
```

Comment:
are the CSS classes automatically added to the anchor?

Reply:
@ralucaStan yes they are. Ariakit is a bit magic but it merges all the props passed onto the component passed to `render`

Reply:
Nice!